src/sqlfluff/utils/reflow/sequence.py

```diff
@@ -458,3 +627,322 @@ def respace(
+        # our range accordingly.
+        for idx in range(2, len(element_buffer) - 2):
+            # Only evaluate blocks:
+            elem = element_buffer[idx]
+            # Only evaluate blocks
+            if not isinstance(elem, ReflowBlock):
+                continue
+            # Does the element itself have config? (The easy case)
+            if elem.line_position:
+                # Blocks should only have one segment so it's easy to pick it.
+                spans.append(
+                    _RebreakSpan(
+                        elem.segments[0],
+                        idx,
+                        idx,
+                        # NOTE: this isn't pretty but until it needs to be more
+                        # complex, this works.
+                        elem.line_position.split(":")[0],
+                        elem.line_position.endswith("strict"),
+                    )
+                )
+            # Do any of its parents have config, and are we at the start
+            # of them?
+            for key in elem.line_position_configs.keys():
+                # If we're not at the start of the segment, then pass.
+                if elem.depth_info.stack_positions[key].idx != 0:
+                    continue
+                # Can we find the end?
+                for end_idx in range(idx, len(element_buffer) - 2):
+                    end_elem = element_buffer[end_idx]
+                    if not isinstance(end_elem, ReflowBlock):
+                        continue
+                    if end_elem.depth_info.stack_positions[key].type in ("end", "solo"):
+                        # Found the end. Add it to the stack.
+                        # We reference the appropriate element from the parent stack.
+                        target_depth = elem.depth_info.stack_hashes.index(key)
+                        target = root_segment.path_to(element_buffer[idx].segments[0])[
+                            target_depth
+                        ].segment
+                        spans.append(
+                            _RebreakSpan(
+                                target,
+                                idx,
+                                end_idx,
+                                # NOTE: this isn't pretty but until it needs to be more
+                                # complex, this works.
+                                elem.line_position_configs[key].split(":")[0],
+                                elem.line_position_configs[key].endswith("strict"),
+                            )
+                        )
+                        break
+                # If we find the start, but not the end, it's not a problem, but
+                # we won't be rebreaking this span. This is important so that we
+                # don't rebreak part of something without the context of what's
+                # in the rest of it. We continue without adding it to the buffer.
+        return spans
```

Comment:
A "bigger picture" question occurred to me as I was reading this function. Are there places in this new code where it would be helpful to introduce preconditions and/or postconditions? I.e. places where we check that things are in the expected state? Kind of analogous to the internal linter check that reruns the parser on "fixed" areas of the code to verify no parse errors were introduced.

In reflow, these checks might enforce things like:
* The alternating structure of a reflow sequence (whitespace, not whitespace)
* No adjacent whitespace segments (is that a rule?)

If we aren't already doing this, a little of this, applied strategically, could help catch bugs and aid in future maintenance.

Reply:
So the first of these (the alternating structure), is checked on every construction of a `ReflowSequence`. That definitely is already enforced.

The second one - that's not really a rule, as much as an easy fix. It's picked up in `respace`, every time that's called, and so I'm not sure the others need to handle it. It might still be worth having it though 🤔 .

Reply:
It's up to you. These were just examples, so there may be other situations where it'd be more valuable. Just a thought/question, don't worry too much about it -- can always add later if we see a need.

Reply:
So the existing `_validate_reflow_sequence` method checks for alternating points and blocks. I've added a clause in there to check for double whitespace, because I agree that should never happen.

Reply:
I take it back - that check causes issues, because I'm using `respace` to _fix_ that when it happens, so I have to accept things like that initially. Good idea, but I'm going to leave it out for now.

Reply:
Sounds good. Perhaps later (if it seems worthwhile), we can devise a way to check the parse tree after reflow-related fixes have been applied. Difficult to check prior to that.

Maybe won't be needed, just remembering last year, finding and fixing little issues left over when fixes were slightly wrong. 🙏